src/encoded/audit/experiment.py

```diff
@@ -4018,6 +4018,45 @@ def audit_experiment_no_processed_data(value, system, files_structure):
         yield AuditFailure('lacking processed data', detail, level='WARNING')
 
 
+def audit_experiment_inconsistent_analyses_files(value, system, files_structure):
+    processed_data = files_structure.get('processed_data')
+    if processed_data and 'analyses' in value:
+        analysis_outputs = set()
+        for analysis in value['analyses']:
+            for f in analysis['files']:
+                analysis_outputs.add(f)
+        files_not_in_analysis = []
+        for processed_file in processed_data:
+            if processed_file in analysis_outputs:
+                continue
+            if processed_file not in analysis_outputs:
+                files_not_in_analysis.append(processed_file)
+                files_not_in_analysis_links = [audit_link(path_to_text(file), file) for file in files_not_in_analysis]
+                detail = ('Experiment {} '
+                        'contains processed file(s) {} '
+                        'not in an analysis'.format(
+                            audit_link(path_to_text(value['@id']), value['@id']),
+                            ', '.join(files_not_in_analysis_links)
+                        )
+                    )
+                yield AuditFailure('inconsistent analyses files', detail, level='INTERNAL_ACTION')
+        files_not_in_processed_data = []
+        for analysis_file in analysis_outputs:
+            if analysis_file in processed_data:
+                continue
+            if analysis_file not in processed_data:
```

Comment:
@bonitalam please update the variable names. processed_data is a dictionary, so every time you do something like "for processed_file in processed_data" or "if analysis_file in processed_data" - you actually iterating over or checking for the existence of a file_id in processed_data dictionary keys. I propose to make it more clear by renaming in the proper places "processed_file" into "processed_file_id" and similarly "analysis_file" into "analysis_file_id", I would also make it clear that you are traversing keys of a dictionary by using "processed_data.keys()"